api/api.proto

```diff
@@ -600,4 +600,82 @@ message GroupSnapCreateResponse {
   // in: body
   // Required: true
   string error = 2;
-}
\ No newline at end of file
+}
+
+service OpenStorageCluster {
+	// Enumerate lists all the nodes in the cluster.
+	rpc Enumerate(ClusterEnumerateRequest)
+		returns (ClusterEnumerateResponse) {}
+
+	// Inspect the node given a UUID.
+	rpc Inspect(ClusterInspectRequest)
+		returns (ClusterInspectResponse) {}
+}
+
+// StorageNode describes the state of the node
+message StorageNode {
+  // Id of the node
+  string id = 1;
+  // Cpu usage of the node
+  double cpu = 2;
+  // Total memory of the node
+  uint64 mem_total = 3;
+  // Used memory of the node
+  uint64 mem_used = 4;
+  // Free memory of the node
+  uint64 mem_free = 5;
+  // Average load (percentage)
+  int64 avg_load = 6;
+  // Node status
+  Status status = 7;
+  // Generated number of the node (XXX Do we need this exposed?)
+  uint64 gen_number = 8;
+  // List of disks on the node 
+  map<string, StorageResource> disks = 9;
+  // List of storage pools this node supports
+  repeated StoragePool pools = 10;
+  // Management IP
+  string mgmt_ip = 11;
+  // Data IP
+  string data_ip = 12;
+  // Timestamp (XXX What is this?)
+  google.protobuf.Timestamp timestamp = 13;
+  // Start time of this node (XXX Time the node started or time that px started?)
+  google.protobuf.Timestamp start_time = 14;
```

Comment:
There is not really a need of start time. But uptime of a node does make sense. May be we can add that later as it will require some changes in ClusterManager.